### ISSUE TITLE
Sphinx latex doc build failed due to a footnotes 

### ISSUE THREAD

**Issue by shirou** [sphinx-doc/sphinx#2206]
When I want to build sphinx document under the `doc` directory and `make latexpdf`, I got this error.

```
processing sphinx.tex... contents intro tutorial invocation rest markup/index markup/toctree markup/para markup/code markup/inline markup/misc domains builders config intl theming templating extensions ext/autodoc ext/autosummary ext/doctest ext/intersphinx ext/math ext/graphviz ext/inheritance ext/ifconfig ext/coverage ext/todo ext/extlinks ext/viewcode ext/linkcode ext/napoleon extdev/index extdev/tutorial extdev/appapi extdev/envapi extdev/builderapi extdev/markupapi extdev/domainapi extdev/nodes websupport web/quickstart web/api web/searchadapters web/storagebackends faq glossary devguide changes examples authors 
resolving references...
None:None: WARNING: u'any' reference target not found: make json
writing... 
Markup is unsupported in LaTeX:
intl:None: footnotes in float captions are not supported by LaTeX
make: *** [latexpdf] Error 1
```

It seems this error occurred at [latex writer](https://github.com/sphinx-doc/sphinx/blob/master/sphinx/writers/latex.py#L1674), `visit_footnote_reference`.

**Comment by shirou** [comment]
If I remove `UnsupportedError`, just ignore, build success.

```
--- a/sphinx/writers/latex.py
+++ b/sphinx/writers/latex.py
@@ -1671,10 +1671,6 @@ class LaTeXTranslator(nodes.NodeVisitor):
             self.body.append('\\footnotemark[%s]' % num)
             self.termfootnotestack[-1].append(footnode)
         else:
-            if self.in_caption:
-                raise UnsupportedError('%s:%s: footnotes in float captions '
-                                       'are not supported by LaTeX' %
-                                       (self.curfilestack[-1], node.line))
```

And the output is below.

<img width="816" alt="2015-12-30 22 52 39" src="https://cloud.githubusercontent.com/assets/177213/12051518/4f2589dc-af48-11e5-8eb7-5390dbc1cd45.png">

If a footnote inside a figure-caption, a number is attached but footnote is not shown. This behavior is current Sphinx limitation I think.

**Comment by tk0miya** [comment]
Thank you for reporting.
I fixed it at 584f051.
